pkg/sql/schemachanger/scexec/scmutationexec/scmutationexec.go

```diff
@@ -134,8 +138,39 @@ func (m *visitor) RemoveColumnDefaultExpression(
 	return nil
 }
 
+func (m *visitor) RemoveTypeBackRefs(ctx context.Context, op scop.RemoveTypeBackRefs) error {
+	// Remove the descriptors namespaces as the last stage
+	tableDesc, err := m.descs.GetMutableTableByID(ctx, op.TableID)
+	if err != nil {
+		return err
+	}
+	dbDesc, err := m.descs.GetImmutableDatabaseByID(ctx, tableDesc.GetParentID())
+	if err != nil {
+		return err
+	}
+	typeIDs, err := tableDesc.GetAllReferencedTypeIDs(dbDesc, func(id descpb.ID) (catalog.TypeDescriptor, error) {
```

Comment:
Bear in mind you can get all referencing descriptor IDs (types or otherwise) using `tableDesc.GetReferencedDescIDs()`. There's nothing wrong with the way you're doing it, I'm just bringing it up in case we want to generalize this operation to not just tables. In fact perhaps we could go even further and generalize `typedesc.Mutable.RemoveReferencingDescriptorID()` to `catalog.MutableDescriptor`, after all removing back-references seems like a common enough chore. Food for thought.